falco.yaml

```
@@ -43,10 +45,13 @@ environment:
       - llvm16
       - m4
       - make
+      - nlohmann-json
       - openssl-dev
       - patch
       - perl
+      - protobuf
       - protobuf-dev
+      - protoc
```

```suggestion
      - protobuf-dev
```
@cpanato just for troubleshooting. Can we try reverting these lines?

---

These changes dose not effect the build env as if only having `protobuf-dev` will also install `protobuf` and `protoc`
```
apk add protobuf-dev
...
(48/55) Installing libprotobuf (3.27.4-r0)
(49/55) Installing libprotobuf-lite (3.27.4-r0)
(50/55) Installing libprotoc (3.27.4-r0)
(51/55) Installing protobuf (3.27.4-r0)
(52/55) Installing abseil-cpp-log-initialize (20240722.0-r0)
(53/55) Installing protoc (3.27.4-r0)
(54/55) Installing zlib-dev (1.3.1-r4)
(55/55) Installing protobuf-dev (3.27.4-r0)
```

But leaving these as they are right now will also be fine.

---

@debasishbsws you're right! 

Anyway, from the log, both in x86 and arch64 builds, it is failing because cmake can't find protobuf installed on the system (or at least, cmake does not see what it expects).

One possible solution is to enable `-DUSE_BUNDLED_PROTOBUF=True` so that cmake will use the bundled one.

---

The issue with system protobuf should be resolved now. But it seems the actual error we were encountering is still there.
Error log:
```
2024/09/03 07:55:15 INFO [ 16%] Building CXX object libscap/engine/gvisor/CMakeFiles/scap_engine_gvisor_o.dir/fillers.cpp.o
2024/09/03 07:55:15 WARN /home/build/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/userspace/libscap/engine/gvisor/parsers.cpp:31:10: fatal error: driver/syscall_compat_x86_64.h: No such file or directory
2024/09/03 07:55:15 WARN    31 | #include <driver/syscall_compat_x86_64.h>
2024/09/03 07:55:15 WARN       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024/09/03 07:55:15 WARN compilation terminated.
2024/09/03 07:55:15 INFO [ 16%] Building CXX object libscap/engine/gvisor/CMakeFiles/scap_engine_gvisor_o.dir/gvisor.cpp.o
2024/09/03 07:55:15 WARN make[3]: *** [libscap/engine/gvisor/CMakeFiles/scap_engine_gvisor_o.dir/build.make:107: libscap/engine/gvisor/CMakeFiles/scap_engine_gvisor_o.dir/parsers.cpp.o] Error 1
2024/09/03 07:55:15 WARN make[3]: *** Waiting for unfinished jobs....
2024/09/03 07:55:15 INFO [ 16%] [MODERN BPF] Building BPF object: /home/build/build/driver/modern_bpf/eventfd2.bpf.o
2024/09/03 07:55:15 WARN /home/build/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/userspace/libscap/engine/gvisor/gvisor.cpp:25:10: fatal error: libscap/engine/gvisor/gvisor.h: No such file or directory
2024/09/03 07:55:15 WARN    25 | #include <libscap/engine/gvisor/gvisor.h>
2024/09/03 07:55:15 WARN       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024/09/03 07:55:15 WARN compilation terminated.
2024/09/03 07:55:15 INFO [ 17%] Building C object libscap/CMakeFiles/scap_platform_util.dir/scap_fds.c.o
2024/09/03 07:55:15 WARN make[3]: *** [libscap/engine/gvisor/CMakeFiles/scap_engine_gvisor_o.dir/build.make:135: libscap/engine/gvisor/CMakeFiles/scap_engine_gvisor_o.dir/gvisor.cpp.o] Error 1
2024/09/03 07:55:15 INFO [ 17%] Building C object libscap/CMakeFiles/scap_engine_util.dir/scap_engine_util.c.o
2024/09/03 07:55:15 INFO [ 17%] [MODERN BPF] Building BPF object: /home/build/build/driver/modern_bpf/execve.bpf.o
2024/09/03 07:55:15 INFO [ 17%] Building C object libscap/CMakeFiles/scap_engine_util.dir/ringbuffer/devset.c.o
2024/09/03 07:55:15 INFO [ 17%] Building C object libscap/CMakeFiles/scap_event_schema.dir/__/__/driver/dynamic_params_table.c.o
2024/09/03 07:55:15 INFO [ 18%] Building C object libscap/CMakeFiles/scap_engine_util.dir/ringbuffer/ringbuffer.c.o
2024/09/03 07:55:15 INFO [ 19%] [MODERN BPF] Building BPF object: /home/build/build/driver/modern_bpf/execveat.bpf.o
2024/09/03 07:55:15 INFO [ 19%] Building C object libscap/CMakeFiles/scap_event_schema.dir/__/__/driver/event_table.c.o
2024/09/03 07:55:15 INFO [ 19%] Building C object libscap/CMakeFiles/scap_platform_util.dir/scap_iflist.c.o
2024/09/03 07:55:15 INFO [ 19%] [MODERN BPF] Building BPF object: /home/build/build/driver/modern_bpf/fchdir.bpf.o
2024/09/03 07:55:15 INFO [ 19%] [MODERN BPF] Building BPF object: /home/build/build/driver/modern_bpf/fchmod.bpf.o
2024/09/03 07:55:15 INFO [ 19%] Building C object libscap/CMakeFiles/scap_event_schema.dir/__/__/driver/flags_table.c.o
2024/09/03 07:55:15 INFO [ 19%] Building C object libscap/CMakeFiles/scap_platform_util.dir/scap_proc_util.c.o
2024/09/03 07:55:15 INFO [ 19%] [MODERN BPF] Building BPF object: /home/build/build/driver/modern_bpf/fchmodat.bpf.o
2024/09/03 07:55:15 WARN /home/build/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/userspace/libscap/engine/gvisor/fillers.cpp:37:10: fatal error: libscap/engine/gvisor/gvisor.h: No such file or directory
2024/09/03 07:55:15 WARN    37 | #include <libscap/engine/gvisor/gvisor.h>
2024/09/03 07:55:15 WARN       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024/09/03 07:55:15 WARN compilation terminated.
```

---

I asked ChatGPT and it responded like this:


> The file syscall_compat_x86_64.h is likely part of a package or project dependency that is either not installed or not correctly configured. gvisor is a container runtime that can be used by Falco for syscalls, and it seems that the necessary header files for compatibility with x86_64 are missing.

---

protobuf dependency changes are a red herring, downgrades were happening due to incomplete abi transitions which are now resolved via rebuilds in https://github.com/wolfi-dev/os/pull/28452

---

Falco maintainer here, I think something might be off with `BUILD_SHARED_LIBS=ON` as there was a patch changing CMake there. It looks like it's not adding the appropriate include paths to the compiler command line. I'm trying to figure out what happened.

---

Hi @LucaGuerra thanks for your help and guidance on this!  Don't suppose you know if there's a way we can use your patch https://github.com/falcosecurity/libs/pull/2079 in the wolfi build before it's merged?  

I tried applying it in the melange pipeline but it seems running `make falco -j$(nproc)` replaces falcosecurity-libs-repo folder so we loose the changes.

---

Hmm I did just manage to apply the changes while in an interactive CLI and it worked - yay! Will have a play to see if we can get this working non-interactively.

---

Thanks a lot! I made a branch for you to test: https://github.com/falcosecurity/libs/commits/test/shared-lib-fix-0.17.3/ . Basically, in `falcosecurity-libs.cmake` set `FALCOSECURITY_LIBS_VERSION` to the commit sha `7eced66e38280da8b9159bc051937fb7254134a9` . The build will complain about the sha256, set it to the new value displayed in the error and you should be good to go. If it works I'll consult with other maintainers to publish a libs patch release for you.

---

awesome - many thanks @LucaGuerra!  I did have a go at changing those values but it didn't seem to help.  I started to double check everything while running another hacky approach in parallel which did work locally.  I'm going to see if that makes the presubmit build happy but if this approach (latest [commit](https://github.com/wolfi-dev/os/pull/26622/commits/5c06d373b03ca34af824fc4c2d486879825a6c74)) isn't too terrible maybe this will suffice for now?

Otherwise am happy to go back to seeing what I likely did wrong with the `FALCOSECURITY_LIBS_VERSION` approach

---

I'm going to resolve this thread as the patch fix works and is only temporary until the upstream PR merges.  Thanks again @LucaGuerra.